### PR TITLE
Dropdown: Avoid calling 'stopProgation' on escape if the dropdown is not expanded

### PR DESCRIPTION
**Pull request by gggjtgxing**
#### Pull request checklist

- [x] Addresses an existing issue: #777
- [x] Include a change request file if publishing 
- [ ] New feature, bugfix, or enhancement
  - [ ] Includes tests
- [ ] Documentation update

#### Description of changes
Currently _onDropdownKeyDown, always does ev.stopPropagation() even if dropdown doesn't handle the key event. but some key event may trigger some behavior in parent components. 
#### Focus areas to test
Test key event work correctly

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -208,36 +208,53 @@ export class Dropdown extends BaseComponent<IDropdownProps, IDropdownState> {
         this.setState({
           isOpen: !this.state.isOpen
         });
+
+        ev.stopPropagation();
```
Comment (mdahamiwal) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: Would be cleaner with having a flag `eventHandled` and set it appropriately in switch cases. In the end we can stop event propagation if event was not handled.

Comment (gggjtgxing) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: I'll update this with a flag

Thanks for reviewing.

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -203,41 +203,54 @@ export class Dropdown extends BaseComponent<IDropdownProps, IDropdownState> {
...
       case KeyCodes.escape:
-        this.setState({
-          isOpen: false
-        });
+        if (this.state.isOpen) {
+          this.setState({
+            isOpen: false
+          });
+          eventHandled = true;
+        }
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: why not just return early here, like the default case, when the state is not open? Then you don't need to track if the event is handled.

File: common/changes/xingwa-dropdown_2017-01-11-08-14.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "office-ui-fabric-react",
+      "comment": "Remove 'stopPropagation' if dropdown doesn't handle the key event.",
```
Comment (dzearing) [inline common/changes/xingwa-dropdown_2017-01-11-08-14.json]: Can you adjust this comment:

Dropdown: Avoid calling stopPropagation on pressing escape if the dropdown is not expanded.

Also it's a patch, not a minor change.

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx (hunk 2 above)
Comment (gggjtgxing) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: This 'avoid calling stopPropagation' applies to other keys as well for which Dropdown doesn't handle in switch case

File: common/changes/xingwa-dropdown_2017-01-11-08-14.json (hunk 3 above)
Comment (gggjtgxing) [inline common/changes/xingwa-dropdown_2017-01-11-08-14.json]: I'll update this, thank you.

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx (hunk 2 above)
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: Right but those other cases are handled in the `default` case... which returns early.

Comment (gggjtgxing) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: yes, you're right, I updated this with return early, thank you, please help review again.

**Pull request merged**